tanjun/checks.py

```diff
@@ -116,20 +116,22 @@ def _optional_kwargs(
 
 
 class _Check:
-    __slots__ = ("_error_message", "_halt_execution")
+    __slots__ = ("_error_embed", "_error_message", "_halt_execution")
 
     def __init__(
         self,
         error_message: typing.Optional[str],
+        error_embed: typing.Optional[hikari.Embed],
```

Comment:
This should take a lot more than just these, consider allowing them to pass CommandError directly here to specify all the args?